src/parser.rs

```diff
@@ -4190,105 +4252,141 @@ pub fn parse_array(tokens: &[Token], index: &mut usize) -> (ParsedExpression, Op
+                (DeclPattern::Name(ParsedName { name, mutable }), span)
+            }
+            TokenContents::LParen => {
+                *index += 1;
 
-            if *index < tokens.len() {
-                match &tokens[*index].contents {
-                    TokenContents::Colon => {
+                let mut inner_pattern = vec![];
+                while *index < tokens.len() {
+                    if tokens[*index].contents == TokenContents::Comma {
                         *index += 1;
-                    }
-                    _ => {
-                        return (
-                            ParsedVarDecl {
-                                name: name.to_string(),
-                                parsed_type: ParsedType::Empty,
-                                mutable: false,
-                                span: tokens[*index - 1].span,
-                                visibility,
-                            },
-                            None,
-                        );
+                    } else if tokens[*index].contents == TokenContents::RParen {
+                        *index += 1;
+                        break;
+                    } else {
+                        let mutable = parse_mutable(tokens, index);
+                        let (name, _, err) = parse_name(tokens, index);
+                        error = error.or(err);
+
+                        inner_pattern.push(ParsedName { name, mutable })
                     }
                 }
-            } else {
-                return (
-                    ParsedVarDecl {
-                        name: name.to_string(),
-                        parsed_type: ParsedType::Empty,
-                        mutable: false,
-                        span: tokens[*index - 1].span,
-                        visibility,
-                    },
-                    None,
-                );
-            }
 
-            if *index < tokens.len() {
-                let decl_span = tokens[*index - 1].span;
-                let mutable = *index + 1 < tokens.len()
-                    && match &tokens[*index].contents {
-                        TokenContents::Name(name) if name == "mutable" => {
-                            *index += 1;
-                            true
-                        }
-                        _ => false,
-                    };
+                let mut span = token.span;
+                span.end = tokens[*index - 1].span.end;
+                (DeclPattern::TupleDestruct(inner_pattern), span)
+            }
+            _ => {
+                error = error.or(Some(JaktError::ParserError(
+                    "expected name or '(' in variable declaration".to_string(),
+                    token.span,
+                )));
+                (DeclPattern::Empty, token.span)
+            }
+        }
+    } else {
+        return (
+            // why do we have to create a structure even when we fail...this feels like a bad design.
```

Comment:
feels like a weird comment to add in...

Reply:
Oops I meant to delete that one. I was frustrated with the parser.